.github/workflows/build.yml

```diff
@@ -234,18 +241,50 @@ jobs:
       - name: Echo outputs
         if: github.event_name != 'pull_request'
         run: |
-          echo "${{ toJSON(steps.push.outputs) }}"  
+          echo "${{ toJSON(steps.push.outputs) }}"
+
+  sign:
+    needs: build_and_push
+    runs-on: ubuntu-latest
+    if: github.event_name != 'pull_request'
+    permissions:
+      id-token: write
+      packages: write
+    steps:
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+
+      - uses: sigstore/cosign-installer@v3
+
+      - name: Sign the images
+        run: cosign sign -y ${TAGS}
+        env:
+          TAGS: ${{needs.build_and_push.outputs.tags}}
```

Comment:
I feel the easiest way to do this would be to move the sign step back to the build jobs, otherwise we'd need to have some logic to extract matrix information from the job metadata

Reply:
The easiest solution would be to move it back into the build job. 

If we wanted to separate the jobs, it would require a few extra steps:

https://www.google.com/url?sa=t&source=web&rct=j&opi=89978449&url=https://github.com/orgs/community/discussions/26284&ved=2ahUKEwjiiaukrauEAxVzkGoFHXGPA4AQFnoECBsQAQ&usg=AOvVaw3iDi-nAVM0p_-IJrisi69Z